administrator/components/com_content/models/shared.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_content
+ *
+ * @copyright   Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+defined('_JEXEC') or die;
+require_once __DIR__ . '/articles.php';
+/**
+ * About Page Model
+ *
+ * @since  1.6
+ */
+class ContentModelShared extends ContentModelArticles
+{
+	/**
+	 * Constructor.
+	 *
+	 * @param   array  $config  An optional associative array of configuration settings.
+	 *
+	 * @see     JController
+	 * @since   1.6
+	 */
+	public function __construct($config = array())
+	{
+		if (empty($config['filter_fields']))
+		{
+			$config['filter_fields'] = array(
+				'id', 'a.id',
+				'title', 'a.title',
+				'alias', 'a.alias',
+				'checked_out', 'a.checked_out',
+				'checked_out_time', 'a.checked_out_time',
+				'catid', 'a.catid', 'category_title',
+				'state', 'a.state',
+				'access', 'a.access', 'access_level',
+				'created', 'a.created',
+				'created_by', 'a.created_by',
+				'created_by_alias', 'a.created_by_alias',
+				'ordering', 'a.ordering',
+				'featured', 'a.featured',
+				'share','a.share',
+				'language', 'a.language',
+				'hits', 'a.hits',
+				'publish_up', 'a.publish_up',
+				'publish_down', 'a.publish_down',
+				'fp.ordering',
+				'published', 'a.published',
+				'author_id',
+				'category_id',
+				'level',
+				'tag'
+			);
+		}
+		parent::__construct($config);
+	}
+	/**
+	 * Build an SQL query to load the list data.
+	 *
+	 * @param   boolean  $resolveFKs  True to join selected foreign information
+	 *
+	 * @return  string
+	 *
+	 * @since   1.6
+	 */
+	protected function getListQuery($resolveFKs = true)
+	{
+		// Create a new query object.
+		$db = $this->getDbo();
+		$query = $db->getQuery(true);
+		// Select the required fields from the table.
+		$query->select(
```

Comment:
@nikitadhiman This query is not needed in your case. You only have two tables to query. Why are you not writing your own query for the shared table? You are pulling in data here that is not needed at all.